Reference/conformance/conformance.pb.swift

```diff
@@ -2162,67 +2154,51 @@ public struct Conformance_TestAllTypes: ProtobufGeneratedMessage {
     set {_uniqueStorage()._mapStringForeignEnum = newValue}
   }
 
-  public var oneofUint32: UInt32? {
+  public var oneofUint32: UInt32 {
     get {
       if case .oneofUint32(let v) = _storage._oneofField {
         return v
       }
-      return nil
+      return 0
     }
     set {
-      if let newValue = newValue {
-        _uniqueStorage()._oneofField = .oneofUint32(newValue)
-      } else {
-        _uniqueStorage()._oneofField = .None
-      }
+      _uniqueStorage()._oneofField = .oneofUint32(newValue)
     }
   }
 
-  public var oneofNestedMessage: Conformance_TestAllTypes.NestedMessage? {
+  public var oneofNestedMessage: Conformance_TestAllTypes.NestedMessage {
     get {
       if case .oneofNestedMessage(let v) = _storage._oneofField {
         return v
       }
-      return nil
+      return Conformance_TestAllTypes.NestedMessage()
     }
     set {
-      if let newValue = newValue {
-        _uniqueStorage()._oneofField = .oneofNestedMessage(newValue)
-      } else {
-        _uniqueStorage()._oneofField = .None
-      }
+      _uniqueStorage()._oneofField = .oneofNestedMessage(newValue)
     }
   }
 
-  public var oneofString: String? {
+  public var oneofString: String {
```

Comment:
This was intentional; users should rely on the case of the `oneof` to determine which value it has, rather than querying the individual fields. This is a different use case from asking the field directly for its value and returning the default if the case is different, which should still be allowed. They can also set the oneof case property itself to `None` to clear it (although we could debate whether we should also provide a `clear` method for that, since that extra case may not be discoverable enough).

It's worth noting that C++ doesn't provide `has` methods for the individual oneof fields either (the documentation is wrong in this regard).